weasyl/submission.py

```diff
@@ -1053,41 +1051,32 @@ def select_recently_popular():
 
     :return: A list of submission dictionaries, in score-rank order.
     """
-    db = meta.Base.dbsession
-    subq = (
-        db.query(content.Favorite.targetid, sa.func.count().label('faves'))
-        .filter_by(type='s')
-        .group_by(content.Favorite.targetid)
-        .subquery())
-    score = (
-        sa.func.log(sa.func.greatest(sa.func.coalesce(subq.c.faves, 0), 1)) +
-        sa.func.log(sa.func.greatest(content.Submission.page_views, 1)) / 2 +
-        sa.cast(content.Submission.unixtime, sa.types.INTEGER) / 180000).label('score')
-    q = (
-        db.query(content.Submission, users.Profile, score)
-        .options(sa.orm.joinedload('tag_objects'))
-        .filter(~content.Submission.is_hidden, ~content.Submission.is_friends_only)
-        .outerjoin(subq, content.Submission.submitid == subq.c.targetid)
-        .join(users.Profile, content.Submission.userid == users.Profile.userid)
-        .order_by(score.desc()))
-
     max_days = int(d.config_read_setting("popular_max_age_days", "21"))
-    if max_days > 0:
-        q = q.filter(content.Submission.unixtime >
-                     sa.func.extract('EPOCH', sa.func.now() - datetime.timedelta(days=max_days)))
-    q = q.limit(128)
-
-    submissions = [{
-        'contype': 10,
-        'score': sc,
-        'submitid': s.submitid,
-        'title': s.title,
-        'rating': s.rating.code,
-        'subtype': s.subtype,
-        'unixtime': s.unixtime.timestamp,
-        'tags': list(s.tags),
-        'userid': s.userid,
-        'username': p.username,
-    } for s, p, sc in q.all()]
+
+    query = d.engine.execute("""
+        SELECT
+            log(count(favorite.*) + 1) + log(submission.page_views + 1) / 2 + submission.unixtime / 180000 AS score,
```

Comment:
I know the accuracy loss is minor, but how much savings does this have over the max approach?